### DISCUSSION TITLE
ANSI colors under Windows

### DISCUSSION THREAD

**Discussion by knutwannheden** [jbangdev/jbang#1255] [Ideas] [answered]
When I run jbang under Windows I either don't get any ANSI colors (when running using `cmd.exe` or `powershell.exe`) or ANSI escape sequences end up cluttering my screen.

AFAICT Picocli should be able to work properly even on Windows but probably requires Jansi (see https://picocli.info/#_windows).

I couldn't find any Jansi dependency in jbang, but I could try work out a PR, if you think adding this dependency would be acceptable.

/cc @maxandersen

**Comment by quintesse** [comment, marked as answer]
If it is something that is not overly big and can be integrated in the jbang.jar then I don't see any problem with including that dependency. Quite the contrary, it would be nice to see colors on Windows!

**Reply by knutwannheden** [reply]
At the moment the ANSI escape codes just garble the output to the point that it is hard to read, when running under Git Bash, because Picocli's heuristics think that colors should be working.

I will look into it.

**Reply by knutwannheden** [reply]
A simple test confirms that just calling `AnsiConsole.systemInstall()` in `Main#main` fixes the issue.

**Reply by knutwannheden** [reply]
The following PR should address the issue: #1256
